docs/changes/units/15228.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Fixed ``astropy.units.Quantity``'s implementation of ``numpy.nanmedian()`` for
+Numpy >= 1.25
```

Comment:
I think it is good for people to know what the problem was, so maybe add ", where for Numpy >= 1.25 an exception was raised for some array shapes and axis combinations." That way people know there were no silent wrong results!